.github/workflows/ci.yml

```diff
@@ -0,0 +1,148 @@
+name: CI
+
+on:
+  push:
+    branches: [ master ]
+    tags: [ "**" ]
+  pull_request:
+    branches: [ "**" ]
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build:
+    name: ${{ matrix.task.name}} - ${{ matrix.os.name }} ${{ matrix.python.name }}
+    runs-on: ${{ matrix.os.runs-on }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - name: Linux
+            runs-on: ubuntu-latest
+        python:
+          - name: CPython 3.8
+            tox: py38
+            action: 3.8
+        task:
+          - name: Build
+            tox: build
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up ${{ matrix.python.name }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python.action }}
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip setuptools wheel
+        python -m pip install build check-manifest twine
+
+    - uses: twisted/python-info-action@v1
+
+    - name: Build
+      run: |
+        check-manifest --verbose .
+
+        python -m build --sdist --outdir dist/ .
+
+        mkdir empty/
+        cd empty
+
+        tar -xvf ../dist/*
+        cd *
+
+        # build the wheel from the sdist
+        python -m build --wheel --outdir ../../dist/ .
+        cd ../../
+
+        twine check dist/*
+
+    - name: Publish
+      uses: actions/upload-artifact@v2
+      with:
+        name: dist
+        path: dist/
+
+  test:
+    name: ${{ matrix.task.name}} - ${{ matrix.os.name }} ${{ matrix.python.name }}
+    runs-on: ${{ matrix.os.runs-on }}
+    needs:
+      - build
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - name: Linux
+            runs-on: ubuntu-latest
+        python:
+          - name: CPython 3.5
+            tox: py35
+            action: 3.5
+          - name: CPython 3.6
+            tox: py36
+            action: 3.6
+          - name: CPython 3.7
+            tox: py37
+            action: 3.7
+          - name: CPython 3.8
+            tox: py38
+            action: 3.8
+          - name: CPython 3.9
+            tox: py39
+            action: 3.9
+          - name: PyPy 3.6
+            tox: pypy36
+            action: pypy-3.6
+          - name: PyPy 3.7
+            tox: pypy37
+            action: pypy-3.7
+        task:
+          - name: Test
+            tox: tests
```

Comment:
Will this also run `precommit.py`?

Reply:
We could setup whatever to run, sure.  I personally haven't gotten up to working on precommit stuff, just hasn't bubbled up the list...  but yeah.  Maybe make a ticket and list out all the bits you want.  As mentioned, I think it's worth getting the basics in and then filling out the rest.